owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/ShareActivity.kt

```diff
@@ -0,0 +1,385 @@
+        ft.replace(
+            R.id.share_fragment_container, searchFragment,
+            TAG_SEARCH_FRAGMENT
+        )
+        ft.addToBackStack(null)    // BACK button will recover the ShareFragment
+        ft.commit()
+    }
+
+    // Private share creation needs to be handled from here since is is carried out through intents
+    override fun onNewIntent(intent: Intent) {
+        when (intent.action) {
+            Intent.ACTION_SEARCH -> {  // Verify the action and get the query
+                val query = intent.getStringExtra(SearchManager.QUERY)
+                Log_OC.w(TAG, "Ignored Intent requesting to query for $query")
+            }
+            UsersAndGroupsSearchProvider.suggestIntentAction -> {
+                val data = intent.data
+                val dataString = intent.dataString
+                val shareWith = dataString!!.substring(dataString.lastIndexOf('/') + 1)
+                createPrivateShare(
+                    shareWith,
+                    data?.authority
+                )
+            }
+            else -> Log_OC.e(TAG, "Unexpected intent $intent")
+        }
+    }
+
+    private fun createPrivateShare(shareeName: String, dataAuthority: String?) {
+        val shareType = UsersAndGroupsSearchProvider.getShareType(dataAuthority)
+
+        ocShareViewModel.insertPrivateShare(
+            file.remotePath,
+            shareType,
+            shareeName,
+            getAppropiatePermissions(shareType)
+        )
+    }
+
+    private fun observePrivateShareCreation() {
+        ocShareViewModel.privateShareCreationStatus.observe(
+            this,
+            Observer { uiResult ->
+                when (uiResult?.status) {
+                    Status.ERROR -> {
+                        Snackbar.make(
+                            findViewById(android.R.id.content)!!,
+                            uiResult.errorMessage!!,
+                            Snackbar.LENGTH_SHORT
+                        ).show()
+                        dismissLoadingDialog()
+                    }
+                    Status.LOADING -> {
+                        showLoadingDialog(R.string.common_loading)
+                    }
+                    else -> {
+                        Log.d(
+                            TAG, "Unknown status when creating private share"
+                        )
+                    }
+                }
+            }
+        )
+    }
+
+    private fun getAppropiatePermissions(shareType: ShareType?): Int {
+        // check if the Share is FERERATED
+        val isFederated = ShareType.FEDERATED == shareType
+
+        when {
+            file.isSharedWithMe -> return RemoteShare.READ_PERMISSION_FLAG    // minimum permissions
+            isFederated -> {
+                val serverVersion = com.owncloud.android.authentication.AccountUtils.getServerVersion(account)
+                return if (serverVersion != null && serverVersion.isNotReshareableFederatedSupported) {
```

Comment:
use `when` better

Reply:
Even with two conditions? I rather use if else for those cases.